internal/functions/registernotification/register-notification.go

```diff
@@ -137,3 +149,28 @@ func handleForFUID(ctx context.Context, fuid string) error {
 
 	return nil
 }
+
+func parseExpirationTime(idToken string) int64 {
+	token, _ := jwt.Parse(idToken, func(token *jwt.Token) (interface{}, error) {
+		return []byte("hello-world"), nil
+	})
+
+	// We certainly got validation error but we don't care. It's just about whether we are able to parse the token or not.
+
+	if token == nil || token.Claims == nil {
+		return 0
+	}
+
+	claims := token.Claims.(jwt.MapClaims)
+
+	switch exp := claims["exp"].(type) {
+	case float64:
+		return int64(exp)
+	case json.Number:
+		v, _ := exp.Int64()
+		return v
+	}
```

Comment:
looks much better, but could we maybe add default: with error log "Unknown type %T" or something like that? :D